integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonSessionState.scala

```diff
@@ -212,6 +232,21 @@ class CarbonOptimizer(
             PredicateSubquery(tPlan, p.children, p.nullAware, p.exprId)
         }
     }
-    super.execute(transFormedPlan)
+    transFormedPlan
+  }
+}
+
+object SessionStateFactory {
+  def getSessionState(sparkSession: SparkSession, className: String): HiveSessionState = {
```

Comment:
Add comment for this function

Reply:
I think you can add this func in CarbonSession directly instead of creating a new object